File: CommonLibrary/src/main/java/com/common/widget/SimpleMarkdownView.java
package com.common.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.webkit.WebChromeClient;
import android.webkit.WebView;
import android.widget.ProgressBar;

import com.mukesh.MarkdownView;

/**
 * by y on 2017/5/17
 */

public class SimpleMarkdownView extends MarkdownView {

    private ProgressBar progressbar;
    private boolean isLoading = false;

    public SimpleMarkdownView(Context context) {
        super(context);
        init();
    }

    public SimpleMarkdownView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public SimpleMarkdownView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {

        progressbar = new ProgressBar(getContext(), null, android.R.attr.progressBarStyleHorizontal);
        progressbar.setLayoutParams(new WebView.LayoutParams(WebView.LayoutParams.MATCH_PARENT, 26, 0, 0));
        addView(progressbar);
        setWebChromeClient(
                new WebChromeClient() {
                    @Override
                    public void onProgressChanged(WebView view, int newProgress) {
                        super.onProgressChanged(view, newProgress);
                        newProgressBar(newProgress);
                    }
                });
    }

    public boolean isLoading() {
        return isLoading;
    }


    private void newProgressBar(int newProgress) {
        if (newProgress == 100) {
            progressbar.setVisibility(GONE);
            isLoading = true;
        } else {
            if (progressbar.getVisibility() == GONE) {
                progressbar.setVisibility(VISIBLE);
            }
            isLoading = false;
            progressbar.setProgress(newProgress);
        }
    }

    @Override
    protected void onScrollChanged(int l, int t, int oldl, int oldt) {
        if (progressbar != null) {
            WebView.LayoutParams lp = (WebView.LayoutParams) progressbar.getLayoutParams();
            lp.x = l;
            lp.y = t;
            progressbar.setLayoutParams(lp);
        }
        super.onScrollChanged(l, t, oldl, oldt);
    }
}

File: app/src/main/java/com/codekk/ui/widget/SimpleMarkdownView.kt
@file:Suppress("DEPRECATION")

package com.codekk.ui.widget

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.webkit.WebChromeClient
import android.webkit.WebView
import android.widget.ProgressBar
import com.mukesh.MarkdownView

/**
 * by y on 2017/5/17
 */
class SimpleMarkdownView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : MarkdownView(context, attrs, defStyleAttr) {

    private var progressbar: ProgressBar = ProgressBar(context, null, android.R.attr.progressBarStyleHorizontal)
    var isLoading = false
        private set

    init {
        progressbar.layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, 26, 0, 0)
        addView(progressbar)
        webChromeClient = object : WebChromeClient() {
            override fun onProgressChanged(view: WebView, newProgress: Int) {
                super.onProgressChanged(view, newProgress)
                newProgressBar(newProgress)
            }
        }
    }

    private fun newProgressBar(newProgress: Int) {
        if (newProgress == 100) {
            progressbar.visibility = View.GONE
            isLoading = true
        } else {
            if (progressbar.visibility == View.GONE) {
                progressbar.visibility = View.VISIBLE
            }
            isLoading = false
            progressbar.progress = newProgress
        }
    }

    override fun onScrollChanged(l: Int, t: Int, oldl: Int, oldt: Int) {
        val lp = progressbar.layoutParams as LayoutParams
        lp.x = l
        lp.y = t
        progressbar.layoutParams = lp
        super.onScrollChanged(l, t, oldl, oldt)
    }
}
